pandas/indexes/multi.py

```diff
@@ -2069,20 +2069,14 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
         else:
 
             loc = level_index.get_loc(key)
-            if level > 0 or self.lexsort_depth == 0:
+            if isinstance(loc, slice):
+                return loc
+            elif level > 0 or self.lexsort_depth == 0:
                 return np.array(labels == loc, dtype=bool)
-            else:
-                # sorted, so can return slice object -> view
-                try:
-                    loc = labels.dtype.type(loc)
```

Comment:
OH I c. ok I'll add that back. The problem I thought was it was a slice ,but you are talking about a type-mismatch, e.g. ``int8`` vs lables of ``int64`` or whatever?